### PR TITLE
[receiver/mongodbatlasreceiver] Add support to collect organization events

### PR DESCRIPTION
**Pull request by JonathanWamsley**
**Description:** <Describe what has changed.>
This PR adds support for collecting organization events from MongoDB Atlas API.

**Link to tracking Issue:** <Issue number if applicable>
#19449, #20308

**Testing:** <Describe what testing was performed and which tests were added.>
Added unit tests

Validated org events working on a live instance.

**Documentation:** <Describe the documentation added.>
Added documentation to the configuration and README

### CODE REVIEW

**Comment by JonathanWamsley** [comment]
@djaglowski Would you mind taking a look at this please. Thank you

File: receiver/mongodbatlasreceiver/config.go [hunk 1]
```diff
@@ -209,8 +215,8 @@ func (a AlertConfig) validateListenConfig() error {
 }
 
 func (e EventsConfig) validate() error {
-	if len(e.Projects) == 0 {
-		return errNoProjects
```
Comment (djaglowski) [inline receiver/mongodbatlasreceiver/config.go:213]: This error is still defined above. Is it still used?

File: receiver/mongodbatlasreceiver/events.go [hunk 2]
```diff
@@ -42,7 +42,9 @@ const (
 
 type eventsClient interface {
 	GetProject(ctx context.Context, groupID string) (*mongodbatlas.Project, error)
-	GetEvents(ctx context.Context, groupID string, opts *internal.GetEventsOptions) (ret []*mongodbatlas.Event, nextPage bool, err error)
+	GetProjectEvents(ctx context.Context, groupID string, opts *internal.GetEventsOptions) (ret []*mongodbatlas.Event, nextPage bool, err error)
+	GetOrganization(ctx context.Context, orgID string) (*mongodbatlas.Organization, error)
+	GetOrgEvents(ctx context.Context, orgID string, opts *internal.GetEventsOptions) (ret []*mongodbatlas.Event, nextPage bool, err error)
```
Comment (djaglowski) [inline receiver/mongodbatlasreceiver/events.go:47]:
```suggestion
	GetOrganizationEvents(ctx context.Context, orgID string, opts *internal.GetEventsOptions) (ret []*mongodbatlas.Event, nextPage bool, err error)
```

File: receiver/mongodbatlasreceiver/events.go [hunk 3]
```diff
@@ -219,6 +261,44 @@ func (er *eventsReceiver) transformEvents(now pcommon.Timestamp, events []*mongo
 	return logs
 }
 
+func (er *eventsReceiver) transformOrgEvents(now pcommon.Timestamp, events []*mongodbatlas.Event, o *mongodbatlas.Organization) plog.Logs {
```
Comment (djaglowski) [inline receiver/mongodbatlasreceiver/events.go:264]: This function is almost exactly the same as transformProjectEvents. Why not combine them?

File: receiver/mongodbatlasreceiver/testdata/events/sample-payloads/org-events.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                "href": "https://cloud.mongodb.com/api/atlas/v1.0/orgs/5b478b3afc4625789ce616a3/events/5b43d04087d9d6357de591af",
+                "rel": "self"
+            }
+        ],
+        "orgId": "5b478b3afc4625789ce616a3",
+        "remoteAddress": "192.0.2.88",
+        "userId": "5898b79080eef53b3ad04e68",
+        "username": "j.doe@example.com"
+    }
+]
```
Comment (djaglowski) [inline receiver/mongodbatlasreceiver/testdata/events/sample-payloads/org-events.json:36]:
```suggestion
]

```

File: receiver/mongodbatlasreceiver/testdata/events/golden/project-events.json [hunk 5]
```diff
@@ -0,0 +1,228 @@
+{
```
Comment (djaglowski) [inline receiver/mongodbatlasreceiver/testdata/events/golden/project-events.json:1]: Golden files should be yaml only at this point. Looks like you may have duplicates in your PR

File: receiver/mongodbatlasreceiver/internal/mongodb_atlas_client.go [hunk 6]
```diff
@@ -687,6 +687,30 @@ func (s *MongoDBAtlasClient) GetEvents(ctx context.Context, groupID string, opts
 	return events.Results, hasNext(response.Links), nil
 }
 
+// GetOrgEvents returns the events specified for the set organizations
+func (s *MongoDBAtlasClient) GetOrgEvents(ctx context.Context, orgID string, opts *GetEventsOptions) (ret []*mongodbatlas.Event, nextPage bool, err error) {
```
Comment (djaglowski) [inline receiver/mongodbatlasreceiver/internal/mongodb_atlas_client.go:691]: Projects are associated with an organization, right? If I were a user, I might expect that getting "organization events" means getting "all events from all projects in an organization". I think we should have some language in the README at least to clarify that these are "events not associated with a project" or something to that effect.

File: receiver/mongodbatlasreceiver/README.md [hunk 7]
```diff
@@ -59,7 +60,9 @@ MongoDB Atlas [Documentation](https://www.mongodb.com/docs/atlas/reference/api/l
     - `exclude_clusters` (default empty)
 - `events`
   - `projects`
-    - `name` Name of the Project to discover events from
+    - `name` Name of the Project to discover events from.
+  - `orgs`
```
[the remark this answers is not in the document]
Comment (JonathanWamsley) [reply receiver/mongodbatlasreceiver/README.md]: Updated orgs to organizations

File: receiver/mongodbatlasreceiver/config.go (hunk 1 above)
Comment (JonathanWamsley) [reply receiver/mongodbatlasreceiver/config.go:213]: Yeah it is still used for the [alert](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/mongodbatlasreceiver/config.go#L166) and [log validation](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/mongodbatlasreceiver/config.go#L136)

File: receiver/mongodbatlasreceiver/events.go (hunk 2 above)
Comment (JonathanWamsley) [reply receiver/mongodbatlasreceiver/events.go]: updated

File: receiver/mongodbatlasreceiver/testdata/events/sample-payloads/org-events.json (hunk 4 above)
Comment (JonathanWamsley) [reply receiver/mongodbatlasreceiver/testdata/events/sample-payloads/org-events.json]: updated

File: receiver/mongodbatlasreceiver/testdata/events/golden/project-events.json (hunk 5 above)
Comment (JonathanWamsley) [reply receiver/mongodbatlasreceiver/testdata/events/golden/project-events.json]: removed

File: receiver/mongodbatlasreceiver/internal/mongodb_atlas_client.go (hunk 6 above)
Comment (JonathanWamsley) [reply receiver/mongodbatlasreceiver/internal/mongodb_atlas_client.go]: An organization can have many projects, so a project is under an organization. From my understanding and testing the endpoints, these organization events won't capture project specific events. Organization events deal with API keys, billing charges, and groups changes. Project events are related to host, cluster and alerts from inspection. I updated the readme.

File: receiver/mongodbatlasreceiver/events.go (hunk 3 above)
Comment (JonathanWamsley) [reply receiver/mongodbatlasreceiver/events.go:235]: updated

File: receiver/mongodbatlasreceiver/README.md [hunk 8]
```diff
@@ -19,6 +20,10 @@ below both values are being pulled from the environment.
 
 In order to collect logs, at least one project must be specified. By default, logs for all clusters within a project will be collected. Clusters can be limited using either the `include_clusters` or `exclude_clusters` setting.
 
+In order to collect project events, the requesting API key needs the appropriate permission which at minimum is the `Project Read Only` role. Project events are specific to a single project.
+
+In order to collect organization events, the requesting API needs the appropriate permission which at minimum is the `Organization Member` role. Organization events are collected across all the projects hosted on Atlas within the organization. These events are not associated with a project.
```
Comment (djaglowski) [inline receiver/mongodbatlasreceiver/README.md:25]: Should this say "API key"

File: receiver/mongodbatlasreceiver/config.go [hunk 9]
```diff
@@ -111,6 +116,7 @@ var (
 
 	// Logs Receiver Errors
 	errNoProjects    = errors.New("at least one 'project' must be specified")
+	errNoEvents      = errors.New("at least one 'project' or 'org' event type must be specified")
```
Comment (djaglowski) [inline receiver/mongodbatlasreceiver/config.go:119]:
```suggestion
	errNoEvents      = errors.New("at least one 'project' or 'organization' event type must be specified")
```

**Review by djaglowski — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Pull request merged**